httpload/cmd/httpload/main_test.go

```diff
@@ -35,3 +36,30 @@ func TestNumberOfRequests(t *testing.T) {
 		t.Fatal("failed to fulfill requests, want:", want, "got:", counter)
 	}
 }
+
+func TestNumberOfRequestsPerSecondPerWorker(t *testing.T) {
+	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(200)
+		w.Write([]byte("Success!"))
+	}))
+	defer srv.Close()
+
+	flags := flag.NewFlagSet("httpload test", flag.ExitOnError)
+	flags.Usage = func() {
+		progName := os.Args[0]
+		fmt.Fprintf(flags.Output(), "Usage: %s [options...] URL\n", progName)
+		flags.PrintDefaults()
+	}
+
+	args := []string{"-w", "1", "-n", "100", "-q", "25", srv.URL}
+
+	start := time.Now()
+	trueMain(flags, args)
+	elapsed := time.Since(start)
+
+	var maxError time.Duration = 1 * time.Second
+	var want time.Duration = 4*time.Second + maxError
+	if elapsed > want {
+		t.Fatal("failed to execute in allowed time, want:", want, "got:", elapsed)
+	}
+}
```

Comment:
I think we could provide both min and max limits to the `elapsed` time here. We must not perform more than 25 rps (requests per second) and we must run 100 queries, that means that the elapsed time must not be under 4 seconds, or it would mean the rps is not respected.
We are indeed testing that we are performing